app/controllers/users/edit_password_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Users
+  class EditPasswordController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def edit
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+    end
+
+    def update
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+
+      result = @update_user_password_form.submit(user_params)
+
+      analytics.track_event(:password_change, result)
+
+      if result[:success?]
+        handle_success
+      else
+        render :edit
+      end
+    end
+
+    private
+
+    def user_params
+      params.require(:update_user_password_form).permit(:password, :current_password)
+    end
+
+    def handle_success
+      bypass_sign_in current_user
```

Comment:
I'm not 100% sure. Devise does it in their user profile update controllers, so I thought we needed it.